lib/puppet/pops/lookup/module_data_provider.rb

```diff
@@ -47,14 +47,12 @@ def key_lookup_in_default(key, lookup_invocation, merge)
   def validate_data_hash(data_hash)
     super
     module_prefix = "#{module_name}::"
-    data_hash.each_key.reduce(data_hash) do |memo, k|
+    data_hash.each_key.each_with_object(data_hash) do |k, memo|
```

Comment:
2 kind of "nitty" points here:

1. Would it make more sense to use `keys` instead of `each_key`? Since we aren't passing a block to `each_key`, it might be more readable.
2. I know that the term `memo` is used here before your PR, but AFAICT there isn't any memoization going on, ie caching a value so it is not computed later. I think I would prefer the term `hash` instead.

Reply:
I've updated my PR with your suggested changes